File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int i = 0;
        int j = numbers.length-1;
        int[] answer = new int[2];
        while(i<j){
            if(numbers[i]+numbers[j]<target){
                i++;
            }
            else if(numbers[i]+numbers[j] == target){
                answer[0] = i+1;
                answer[1] = j+1;
                return answer;
            }
            else{
                j--;
            }
        }
        return answer;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if(root == nullptr){
            return {};
        }
        int max = 1;
        std::queue<TreeNode*> que;
        que.push(root);
        std::vector<vector<int>> answer;
        std::vector<int> holder;
        std::vector<TreeNode*> children;
        while(!que.empty()){
            while(!que.empty()){
                TreeNode* current = que.front();
                if(current->left != nullptr){
                    children.push_back(current->left); 
                }  
                if(current->right != nullptr){
                    children.push_back(current->right); 
                }
                holder.push_back(current->val);
                que.pop();
            }
            for(TreeNode* i: children){
                que.push(i);
            }
            answer.push_back(holder);
            holder = {};
            children = {};
        }
        if(holder.size()>0){
            answer.push_back(holder);
        }
        return answer;

    }
};
